backend/src/test/repository/DiscountRepository.test.ts

```diff
@@ -1,20 +1,23 @@
-import { getConnection, getCustomRepository } from 'typeorm';
+import { getCustomRepository } from 'typeorm';
 import { discounts_sample } from '../../main/resources/Data';
-import { BaseRepositoryTest } from './BaseRepositoryTest';
+import connection from './BaseRepositoryTest';
 import { Discount } from '../../main/entity/Discount';
 import { DiscountRepository } from '../../main/repository/DiscountRepository';
 
 describe('Unit tests for DiscountRepository', function () {
   let discountRepository: DiscountRepository;
-  beforeEach(() => {
-    return BaseRepositoryTest.establishTestConnection().then(() => {
-      discountRepository = getCustomRepository(DiscountRepository);
-    });
+
+  beforeAll(async () => {
+    await connection.create();
+  });
+
+  afterAll(async () => {
+    await connection.close();
   });
 
-  afterEach(() => {
-    const conn = getConnection();
-    return conn.close();
+  beforeEach(async () => {
+    await connection.clear();
+    discountRepository = getCustomRepository(DiscountRepository);
   });
```

Comment:
So before we run the whole test class, we create a connection, before each test we clear the tables and set the repositories, after all tests run, we clear the tables and close the connection.

**Note** we need to clear the tables after all tests run so that no data is left behind in foodies_test